widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt

```diff
@@ -446,9 +453,13 @@ internal class ChatController(
         if (chatState.isOperatorOnline) dialogController.showExitChatDialog(chatState.formattedOperatorName)
     }
 
-    fun leaveChatQueueClicked() {
-        Logger.d(TAG, "leaveChatQueueClicked")
-        dialogController.showExitQueueDialog()
+    fun onXButtonClicked() {
+        Logger.d(TAG, "onXButtonClicked")
+        if (isOngoingEngagementUseCase()) {
+            dialogController.showExitQueueDialog()
+        } else {
```

Comment:
I'm a bit confused here. I thought that queueing and engagement are different states. If so, then this logic is confusing. I would expect to see "if **queueing** then showExitQueueDialog".